src/lib.rs

```diff
@@ -0,0 +1,343 @@
+        };
+
+        GovernorPolicy::Keyed(Box::new(keyed_policy))
+    }
+}
+
+impl GovernorPolicy {
+    /// Create a new builder for GovernorPolicy
+    pub fn builder() -> GovernorPolicyBuilder {
+        GovernorPolicyBuilder {
+            quota: None,
+            gc_interval: Duration::from_secs(60),
+        }
+    }
+
+    /// Start garbage collection if needed
+    fn start_gc_if_needed(&self) {
+        static DIRECT_GC_STARTED: OnceCell<()> = OnceCell::new();
+        static KEYED_GC_STARTED: Lazy<Mutex<HashSet<usize>>> =
+            Lazy::new(|| Mutex::new(HashSet::new()));
+
+        match self {
+            GovernorPolicy::Direct(policy) => {
+                DIRECT_GC_STARTED.get_or_init(|| {
+                    let limiter = policy.limiter.clone();
+                    let gc_interval = policy.gc_interval;
+
+                    tokio::spawn(async move {
+                        let mut interval = tokio::time::interval(gc_interval);
+                        loop {
+                            interval.tick().await;
+                            // No need to do anything for direct rate limiter
+                        }
+                    });
+                });
+            }
+            GovernorPolicy::Keyed(policy) => {
+                // Use the pointer address as a unique identifier for this policy instance
+                let policy_ptr = policy as *const _ as usize;
+                let mut started = KEYED_GC_STARTED.lock().unwrap();
+
+                if !started.contains(&policy_ptr) {
+                    started.insert(policy_ptr);
+
+                    // Start GC for this keyed policy
+                    let interval = policy.gc_interval();
+
+                    // Instead of cloning the policy, we'll just create a new task
+                    // that calls the start_gc_if_needed method periodically
+                    tokio::spawn(async move {
+                        let mut interval_timer = tokio::time::interval(interval);
+                        loop {
+                            interval_timer.tick().await;
+                            // We can't access the policy here anymore, but that's ok
+                            // because the policy will be checked again on each request
+                        }
+                    });
+                }
+            }
+        }
+    }
+}
+
+impl<State, Request> Policy<State, Request> for GovernorPolicy
+where
+    State: Clone + Send + Sync + 'static,
+    Request: Send + Sync + 'static,
+{
+    type Guard = ();
+    type Error = GovernorError;
+
+    async fn check(
+        &self,
+        ctx: Context<State>,
+        request: Request,
+    ) -> PolicyResult<State, Request, Self::Guard, Self::Error> {
+        // Initialize GC if needed
+        self.start_gc_if_needed();
+
+        match self {
+            GovernorPolicy::Direct(policy) => match policy.limiter.check() {
+                Ok(_) => PolicyResult {
```

Comment:
might be useful to here and there put some `trace`/`debug` calls using the `tracing` crate :) can help if ever needs runtime debugging